lib/cmd/fh3/app/stage.js

```diff
@@ -1,60 +1,137 @@
+  },{
+    cmd : 'fhc app stage --app=<cloud-app-id> --env=<environment> --runtime=<runtime> --gitRefType=<gitRef-type>',
+    desc : i18n._("Deploy of cloud application <cloud-app-id> on <environment> with the <runtime> with 'branch' or 'tag' informed.")
+  },{
+    cmd : 'fhc app stage --app=<cloud-app-id> --env=<environment> --runtime=<runtime> --gitRefHash=<gitRef-hash>',
+    desc : i18n._("Deploy of cloud application <cloud-app-id> on <environment> with the <runtime> with the commit hash that you'd like to deploy.")
+  },{
+    cmd : 'fhc app stage --app=<cloud-app-id> --env=<environment> --runtime=<runtime> --gitRefValue=<gitRef-value>',
+    desc : i18n._("Deploy of cloud application <cloud-app-id> on <environment> with the <runtime> with the name of the branch that you'd like to deploy.")
+  }],
+  'demand' : ['app', 'env'],
+  'method' : 'post',
+  'alias' : {
+    'app' : 'a',
+    'env' : 'e',
+    'runtime' : 'r',
+    'clean' : 'c',
+    'autodeploy': 'ad',
+    'gitRefType' : 'gt',
+    'gitRefHash' : 'gh',
+    'gitRefValue' : 'gv',
+    0 : 'app',
+    1 : 'env'
+  },
+  'describe' : {
+    'app' : i18n._('Unique 24 character GUID of the cloud app you want to see resource info for'),
+    'env' : i18n._('The cloud environment your app is running in'),
+    'runtime' : i18n._("The Node.js runtime of your application. (E.g. node08, node010 or node4.). To check the available runtimes for this app use '$fhc app runtimes --id=<cloud-app-id>'"),
+    'clean' : i18n._("Options 'true' or 'false'. Do a full, clean stage. Cleans out all old application log files, removes cached node modules and does an 'npm install' from scratch"),
+    'autodeploy' : i18n._("Options 'true' or 'false'. Deploy the app automatically when the app repo is updated"),
+    'gitRefType' : i18n._("Specifies if you'd like to deploy a 'branch' or 'tag'"),
+    'gitRefHash' : i18n._("The commit hash you'd like to deploy if gitRef.type is 'branch'. HEAD will use the latest code for that branch"),
+    'gitRefValue' : i18n._("The name of the branch you'd like to deploy")
+  },
+  'preCmd' : function(params, cb) {
+
+    if (params.runtime) {
+      var domain = ini.get("domain");
+      runtimesCommand.exec({id:params.app, domain: domain}, function(err, data) {
+        if (err) {
+          return cb(i18n._('To found runtimes: ' + err));
+        } else if (!data || !data.result || data.result.empty) {
+          return cb(i18n._('Not found the runtimes for this application.'));
+        }
+        var validRuntimes = [];
+        var found = false;
+        data.result.forEach(function(entry, index) {
+          validRuntimes.push(data.result[index].name);
+          if (data.result[index].name === params.runtime) {
+            found = true;
+          }
+        });
+        if (!found) {
+          return cb(i18n._('Invalid parameter for runtimes!\n The valid options are :' + validRuntimes));
+        }
+      });
+    }
+
+    var data = {
```

Comment:
I think the indentation is not right here. The code from this point should be inside the callback?